persistence/kvstore/kvstore.go

```diff
@@ -0,0 +1,93 @@
+package kvstore
+
+import (
+	"log"
+
+	badger "github.com/dgraph-io/badger/v3"
```

Comment:
Why badger-db?

Leveldb would've been my first choice as far a placeholder considering it's already used in pre-persistence.

Reply:
Discussed offline. Leaving as is for now and we'll figure it out.